src/emulator/cpu.ts

```diff
@@ -395,6 +399,18 @@ export class Cpu {
         return cycles;
     }
 
+    private opCcf(instruction: Instruction): number {
+        this.clock.increment(instruction.cycles);
+
+        // prettier-ignore
+        this.state.r8[r8.f] =
+            (this.state.r8[r8.f] & flag.z) |
```

Comment:
Ein bisschen einfacher: `this.state.r8[r8.f] = (this.state.r8[r8.f] ^ flag.c) & 0x9`

Reply:
Ich kann das gerne ändern, dann sieht man aber nicht mehr genau was gemacht wird. Bringt die Änderung einen Performancezuwachs?

Reply:
Huch, Mensch, und da hab' ich versehentlich Deutsch angefangen 😏 Schneller ist das auf jeden Fall (zwei Logikoperationen vs. drei + ein Conditional), aber wenn das Programm nicht gerade nur aus CCF, dann wird das keinen großen Effekt haben.

An dieser Stelle (in den Opcodes) würde ich im Zweifelsfall immer Performance den Vorrang geben und im Zweifelsfall einen Kommentar schreiben, was das tut. Verständlichkeit ist halt immer auch eine Frage der Perspektive, ich persönlich finde die kurze Variante übersichtlicher und leichter lesbar (C flippen = XOR, N + H maskieren = AND). Ist aber halt alles ein Stück Geschmackssache.

Reply:
Müsste es dann nicht eigentlich 0x90 (anstatt 0x9) heißen? Den die Werte für die Flags sind 0x80 und 0x10, zusammen 0x90.

Reply:
In der Tat, hast rect 😏 Alternativ auch gerne `(flag.z | flag.c)`, ich war nur schreibfaul.

Reply:
Ersetzt der Compiler das dann durch den konstanten Wert 0x90 oder ist das dann wieder ein Logikoperation in der Runtime.

Reply:
> Christian Speckner  21.04.2022 um 13:37 Uhr
> @Valgard Hast Du zufällig geforcepusht? Ich kann an der Geschichte mit den Flags nicht mehr kommentieren. 
> Auf jeden Fall die Antwort: flag.c | flag.z ist von der Performance her identisch zu 0x90 --- 
> Typescript übersetzt die Members const enum in Literale, und jede vernünftige VM berechnet das 
> dann statisch beim Parsen oder Optimieren.